src/thread.cpp

```diff
@@ -229,13 +227,23 @@ Thread* ThreadPool::get_best_thread() const {
         votes[th->worker->rootMoves[0].pv[0]] += thread_value(th);
 
     for (Thread* th : threads)
-        if (std::abs(bestThread->worker->rootMoves[0].score) >= VALUE_TB_WIN_IN_MAX_PLY)
+        if (bestThread->worker->rootMoves[0].score >= VALUE_TB_WIN_IN_MAX_PLY)
         {
-            // Make sure we pick the shortest mate / TB conversion or stave off mate the longest
+            // Make sure we pick the shortest mate / TB conversion
             if (th->worker->rootMoves[0].score > bestThread->worker->rootMoves[0].score)
                 bestThread = th;
         }
+        else if (bestThread->worker->rootMoves[0].score != -VALUE_INFINITE
+                 && bestThread->worker->rootMoves[0].score <= VALUE_TB_LOSS_IN_MAX_PLY)
+        {
+            // Make sure we pick the shortest mated / TB conversion
+            if (th->worker->rootMoves[0].score != -VALUE_INFINITE
+                && th->worker->rootMoves[0].score < bestThread->worker->rootMoves[0].score)
+                bestThread = th;
+        }
         else if (th->worker->rootMoves[0].score >= VALUE_TB_WIN_IN_MAX_PLY
```

Comment:
we should really consider rewriting this (in the future), it has become completely unreadable